### DISCUSSION TITLE
Support X27 stepper motors that can be wired directly to Arduino

### DISCUSSION THREAD

**Discussion by tigert** [MobiFlight/MobiFlight-Connector#636] [Ideas]
Small automotive X27 stepper motors can be wired directly to four pins on arduinos, and they seem to work well for small gauges with lightweight pointer needles. 

https://photos.app.goo.gl/KjCK5Hujxufha34L9

This library works pretty well for this: https://github.com/clearwater/SwitecX25

The current stepper implementation in MobiFlight seems to work differently, and does produce erratic movements, but the above arduino library works very well. These steppers would suit well for small gauges etc. There are even dual concentric versions that you could build

![image](https://user-images.githubusercontent.com/2587818/147885131-037755b7-3fb6-44d7-8d0c-7a1b60bdec51.png)

**Comment by tigert** [comment]
Actually this works already, just had to wire correctly. Wiki tutorial coming soon.

**Comment by nano584658** [comment]
Waiting for tutorial :)

**Reply by DocMoebiuz** [reply]
stepper tutorial will be coming with the next release and all the cool new stepper improvements

**Reply by nano584658** [reply]
Thank you!

**Comment by tigert** [comment]
And there is a wiki page that I did, but forgot to link here, @nano584658 

https://github.com/MobiFlight/MobiFlight-Connector/wiki/How-to-Directly-Drive-x.27-or-x.40-Stepper-Motors

**Reply by nano584658** [reply]
Many thanks @tigert !!

**Reply by tigert** [reply]
And of course that wiki page does not yet have the new finer grained settings for different stepper motor types, but the basic wiring information is what you likely are after anyway. So just swap the middle two wires around if the X.27 doesn't turn normally but just twitches.

**Reply by nano584658** [reply]
That's exactly what's happening to me.  The strange thing is that by connecting the x27 to the mega via the easydriver card, the motor rotates correctly but the movements are extremely slow and not very smooth.
